packages/cli/src/controllers/me.controller.ts

```diff
@@ -231,34 +233,44 @@ export class MeController {
 	}
 
 	/**
-	 * Creates an API Key
+	 * Create an API Key
 	 */
-	@Post('/api-key', { middlewares: [isApiEnabledMiddleware] })
+	@Post('/api-keys', { middlewares: [isApiEnabledMiddleware] })
 	async createAPIKey(req: AuthenticatedRequest) {
 		const apiKey = `n8n_api_${randomBytes(40).toString('hex')}`;
```

Comment:
IMO we should switch to using JWT for API keys as well. Having multiple types of auth tokens is unnecessary maintenance work, and leads to inconsistent semantics across the codebase.

Reply:
You make a good point. When we started working on the api the auth was a jwt, but I do not remember why we decided to change it. 

If we do so:

1. Should we sign the tokens with the same secret as the as the logic tokens?
2. I guess we then do not need the new api keys table.
3. When do we deprecate the existing tokens in the user’s table? n8n v2?

Reply:
> 1. Should we sign the tokens with the same secret as the as the logic tokens?

IMO, Yes.

> 2. I guess we then do not need the new api keys table.
We still need a table to track deleted tokens. So, I prefer that we keep the table as it is, and use the JWT only for expiration and some minimal metadata.

> * When do we deprecate the existing tokens in the user’s table? n8n v2?
Yes, We could start by keeping all the old keys, not validating if they are JWT, and then dropping then in v2.

Reply:
1 and 2 are clear.

With regards to 3. Just to make sure I get it correctly! If that is the case then we won't be able to create the new tokens (jwt) until n8n v2 when we deprecate the old tokens? Right off the top of my head, I was thinking on temporarily validating both types of tokens depending on the header key they come (**x-n8n-api-key**=value or **authorization**=jwt value). We could even keep the same header as I can know which token is which by the shape. From the changes on the users will be able to create only api keys as jwt, but the old ones will be valid until n8n v2, when we can delete them from the api keys table (they all start with n8n_api). Also we could user know after deprecation that they are using an invalid token if they make a request with the old header X-N8N-API-KEY or the token has the old shape.

Reply:
@netroy for whenever you have the chance 🙏